scripts/pages/zoning/zoning.ts

```diff
@@ -0,0 +1,777 @@
+/**
+ * Zoning UI logic
+ */
+
+const TracklistSnippet = {
+	TRACK: 'tracklist-track',
+	SEGMENT: 'tracklist-segment',
+	CHECKPOINT: 'tracklist-checkpoint'
+};
+
+const DefragFlags = {
+	HASTE: 1 << 0,
+	SLICK: 1 << 1,
+	DAMAGEBOOST: 1 << 2,
+	ROCKETS: 1 << 3,
+	PLASMA: 1 << 4,
+	BFG: 1 << 5
+};
+
+class ZoneMenu {
+	static panels = {
+		/** @type {Panel} @static */
+		zoningMenu: $.GetContextPanel(),
+		/** @type {Panel} @static */
+		trackList: $('#TrackList') as Panel,
+		/** @type {Panel} @static */
+		propertiesTrack: $('#TrackProperties') as Panel,
+		/** @type {Panel} @static */
+		defragFlags: $('#DefragFlags') as Panel,
+		/** @type {Panel} @static */
+		propertiesSegment: $('#SegmentProperties') as Panel,
+		/** @type {Panel} @static */
+		propertiesZone: $('#ZoneProperties') as Panel,
+		/** @type {DropDown} @static */
+		filterSelect: $('#FilterSelect') as DropDown,
+		/** @type {DropDown} @static */
+		volumeSelect: $('#VolumeSelect') as DropDown,
+		/** @type {DropDown} @static */
+		regionSelect: $('#RegionSelect') as DropDown,
+		/** @type {Panel} @static */
+		pointsSection: $('#PointsSection') as Panel,
+		/** @type {Panel} @static */
+		pointsList: $('#PointsList') as Panel,
+		/** @type {Panel} @static */
+		propertiesSection: $('#PropertiesSection') as Panel,
+		/** @type {Panel} @static */
+		teleportSection: $('#TeleportSection') as Panel,
+		/** @type {TextEntry} @static */
+		regionBottom: $('#RegionBottom') as TextEntry,
+		/** @type {TextEntry} @static */
+		regionHeight: $('#RegionHeight') as TextEntry,
+		/** @type {TextEntry} @static */
+		regionSafeHeight: $('#RegionSafeHeight') as TextEntry,
+		/** @type {DropDown} @static */
+		regionTPDest: $('#RegionTPDest') as DropDown,
+		regionTPPos: {
+			/** @type {TextEntry} @static */
+			x: $('#TeleX') as TextEntry,
```

Comment:
Just taking a nose at this and wanted to point at right away that you can do `$<TextEntry>('#TeleX')`. Same for allll of these panels, then you don't need any jsdoc annotations.